### PR TITLE
Add RootHash to L1BlockInfo

### PR DESCRIPTION
**Pull request by Inkvi**
- The change is necessary for IBC clients so they can verify consensus updates

## Summary by CodeRabbit

- **New Features**
  - Enhanced block information tracking with the addition of a new RootHash field to improve data integrity and verification processes.

- **Improvements**
  - Updated data serialization and deserialization methods to accommodate the new RootHash field, ensuring seamless integration with existing systems.

### CODE REVIEW

**Comment by dshiell** [comment]
@coderabbitai review

**Review by dshiell — COMMENTED** [review]
Looks like one possible coderabbit issue?

**Comment by nicopernas** [comment]
Please change the base to the `polymer/v1.2.0-dev` branch.

**Comment by Inkvi** [comment]
> Please change the base to the `polymer/v1.2.0-dev` branch.

There is no branch like that in the repo. Did you mean `polymer/v1.2.0`?

File: op-node/rollup/derive/l1_block_info.go [hunk 1]
```diff
@@ -90,6 +92,9 @@ func (info *L1BlockInfo) MarshalBinary() ([]byte, error) {
 	if err := solabi.WriteEthBytes32(w, info.L1FeeScalar); err != nil {
 		return nil, err
 	}
+	if err := solabi.WriteHash(w, info.RootHash); err != nil {
+		return nil, err
+	}
 	return w.Bytes(), nil
 }
```
[the remark this answers is not in the document]
Comment (Inkvi) [reply op-node/rollup/derive/l1_block_info.go:100]: Good catch @coderabbitai!
Update L1InfoArguments from 8 to 9.

**Comment by dshiell** [comment]
> > Please change the base to the `polymer/v1.2.0-dev` branch.
> 
> There is no branch like that in the repo. Did you mean `polymer/v1.2.0`?

Yeah I'm 99% sure that's the correct one to merge with. @nicopernas can confirm.

**Comment by nicopernas** [comment]
> > > Please change the base to the `polymer/v1.2.0-dev` branch.
> 
> > 
> 
> > There is no branch like that in the repo. Did you mean `polymer/v1.2.0`?
> 
> 
> 
> Yeah I'm 99% sure that's the correct one (polymer/v1.2.0) to merge with. @nicopernas can confirm.

Yes! I meant `polymer/v1.2.0`. I messed up :)

**Comment by Inkvi** [comment]
> > > > Please change the base to the `polymer/v1.2.0-dev` branch.
> > 
> > 
> > > 
> > 
> > 
> > > There is no branch like that in the repo. Did you mean `polymer/v1.2.0`?
> > 
> > 
> > Yeah I'm 99% sure that's the correct one (polymer/v1.2.0) to merge with. @nicopernas can confirm.
> 
> Yes! I meant `polymer/v1.2.0`. I messed up :)

I changed the base of the PR and rebased the changes on top of that branch

**Review by jlafiandra6 — APPROVED** [review]
LGTM

**Pull request merged**